### PR TITLE
2-create account

### PR DESCRIPTION
**Pull request by leminhhuan72**
Description:
 - Customer service
 - create account feature in customer service
 - Console UI to print the main menu and handle the create account feature
 - Finish Customer Model
Review:

### CODE REVIEW

File: src/main/java/com/exercise/hotel/model/Customer.java [hunk 1]
```diff
@@ -30,4 +30,14 @@ public String getFirstName() {
...
         return lastName;
     }
+
+    public boolean authenticate(String password) {
+        return this.password.equals(password);
+    }
+    @Override
+    public String toString() {
+        return "Customer{" +
+                "email='" + email + '\'' +
```
Comment (gdragon0598) [inline src/main/java/com/exercise/hotel/model/Customer.java:40]: it should show email + firstName + lastName

File: src/main/java/com/exercise/hotel/model/Customer.java
Comment (gdragon0598) [inline src/main/java/com/exercise/hotel/model/Customer.java:1]: need override method equals... something like: 
` if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Customer customer = (Customer) o;

        return email.equals(customer.email);`

File: src/main/java/com/exercise/hotel/ui/MainMenuService.java
Comment (gdragon0598) [inline src/main/java/com/exercise/hotel/ui/MainMenuService.java:1]: should extend MenuService class

Comment (gdragon0598) [reply src/main/java/com/exercise/hotel/ui/MainMenuService.java:1]: AdminMenuService & MainMenuService extend MenuService(abstract class)
In Menuservice have abstract method:
public abstract void showMenu();

And override in sub-class

File: src/main/java/com/exercise/hotel/ui/MainMenuService.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+
+public class MainMenuService{
+    private final ConsolePrinter consolePrinter;
+    private final BufferedReader bufferedReader;
+    public MainMenuService(ConsolePrinter consolePrinter, BufferedReader bufferedReader) {
+        this.consolePrinter = consolePrinter;
+        this.bufferedReader = bufferedReader;
+    }
+    public void showMenu() {
```
Comment (gdragon0598) [inline src/main/java/com/exercise/hotel/ui/MainMenuService.java:17]: need override

File: src/main/java/com/exercise/hotel/model/Customer.java [hunk 3]
```diff
@@ -3,7 +3,7 @@
 import com.exercise.hotel.util.CustomerValidate;
 
 public class Customer {
```
Comment (gdragon0598) [inline src/main/java/com/exercise/hotel/model/Customer.java:5]: create constructor for this class:
`public Customer(String firstName, String lastName, String email) {
        if (isValidEmail(email)) {
            this.email = email;
        } else {
            throw new IllegalArgumentException("Email is of wrong format. " +
                    "Please correct your email.");
        }
        this.firstName = firstName;
        this.lastName = lastName;
    }`

File: src/main/java/com/exercise/hotel/ui/MenuManager.java [hunk 4]
```diff
@@ -0,0 +1,4 @@
+package com.exercise.hotel.ui;
+
+public interface MenuManager {
```
Comment (gdragon0598) [inline src/main/java/com/exercise/hotel/ui/MenuManager.java:3]: add method   void open(); in this inteface

File: src/main/java/com/exercise/hotel/ui/MainMenuManager.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+public class MainMenuManager implements MenuManager{
+    private final MainMenuService mmService;
+    private final CustomerService customerService;
+    public MainMenuManager(MainMenuService mmService, CustomerService customerService) {
+        this.mmService = mmService;
+        this.customerService = customerService;
+    }
+
+    public void open() throws IOException {
```
Comment (gdragon0598) [inline src/main/java/com/exercise/hotel/ui/MainMenuManager.java:16]: should @ovveride

File: src/main/java/com/exercise/hotel/service/CustomerService.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        customerMap = new HashMap<>();
+    }
+
+    //Method to get the singleton instance
+    public static CustomerService getInstance() {
+        if(csInstance == null)
+            csInstance = new CustomerService();
+        return csInstance;
+    }
+
```
Comment (gdragon0598) [inline src/main/java/com/exercise/hotel/service/CustomerService.java:29]: add comment describe this method:
`/**
     * Creates a new {@link Customer} and records it if no customer already recorded with the provided email.
     *
     * @param email                     string, email of the customer
     * @param firstName                 string, first name of the customer
     * @param lastName                  string, last name of the customer
     * @throws IllegalArgumentException if a customer with the supplied email was already recorded
     */`

File: src/main/java/com/exercise/hotel/service/CustomerService.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    //Simple crud
+    public boolean addCustomer(Customer customer) {
+        if(customerMap.putIfAbsent(customer.getEmail(),customer) == null) {
```
Comment (gdragon0598) [inline src/main/java/com/exercise/hotel/service/CustomerService.java:32]: need check email existed in list customers...if exised throw exception `throw new IllegalArgumentException("Customer with this email is " +
                    "already registered.");`

File: src/main/java/com/exercise/hotel/service/CustomerService.java [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+            return tmp.authenticate(password);
+        }
+    }
+
+    //this is used for test
+    public void showAllCustomer() {
+        List<Customer> list = new ArrayList<Customer>(customerMap.values());
+        list.stream().forEach(System.out::println);
+    }
+
```
Comment (gdragon0598) [inline src/main/java/com/exercise/hotel/service/CustomerService.java:52]: should add new method: getCustomerByEmail(String email)

**Comment by leminhhuan72** [comment]
Em sửa rồi anh check lại

**Comment by leminhhuan72** [comment]
Already fixed, wait for u check it again

**Pull request merged**